pkg/apps/controller/deployer/deployer_controller.go

```diff
@@ -108,6 +108,32 @@ func (c *DeploymentController) handle(deployment *v1.ReplicationController, will
 
 	deployerPodName := deployutil.DeployerPodNameForDeployment(deployment.Name)
 	deployer, deployerErr := c.podLister.Pods(deployment.Namespace).Get(deployerPodName)
+
+	selector := deployutil.DeployerPodSelector(deployment.Name)
+	cm := kcontroller.NewPodControllerRefManager(c.pc, deployment, selector, deployutil.RCControllerRefKind, c.canAdoptFunc(deployment))
+	_, err := cm.ClaimPods([]*v1.Pod{deployer})
+	if err != nil {
+		return err
+	}
+
+	{
```

Comment:
why is this block?

Reply:
to scope variables and not to leak names since this is to be removed after the interim